### ISSUE TITLE
[Bug]: App crashes after entering credentials

### ISSUE THREAD

**Issue by dboehmer** [commons-app/apps-android-commons#5217]
### Summary

I just installed the app from F-Droid, entered my credentials and it just crashes.

APK source: F-Droid link in repo readme

Tried to reboot the phone but didn’t change anything.

### Steps to reproduce

1. Install app
2. Skip introduction screens
3. enter username & password
4. app closes

### Expected behaviour

Login works or fails with an error message but app survives.

### Actual behaviour

App crashes repeatedly. Does so even after clearing the app’s storage. Needed to uninstall and install again to reproduce the behavior.

### Device name

Fairphone 3

### Android version

Android 12 from LineageOS 19

### Commons app version

2.6.0

### Would you like to work on the issue?

Prefer not

**Comment by nicolas-raoul** [comment]
Thanks a lot for reporting!
Would you be able to take a log via USB using "adb logcat"?

**Comment by dboehmer** [comment]
I hope I selected a useful chunk of the log.

```
04-27 14:22:01.093  1528  2453 D CompatibilityChangeReporter: Compat change id reported: 161252188; UID 10145; state: DISABLED
04-27 14:22:01.093  1528  2453 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=fr.free.nrw.commons/.auth.LoginActivity bnds=[21,886][229,1201]} from uid 10091
04-27 14:22:01.103     0     0 I aw8898_startup: enter
04-27 14:22:01.124     0     0 I aw8898_mute: mute state=0
04-27 14:22:01.124     0     0 I         : aw8898_smartpa_cfg, flag = 1
04-27 14:22:01.147  1528  2453 D CompatibilityChangeReporter: Compat change id reported: 194480991; UID 10145; state: ENABLED
04-27 14:22:01.147  1528  2453 D CompatibilityChangeReporter: Compat change id reported: 174042980; UID 10145; state: DISABLED
04-27 14:22:01.149  1528  2453 D CompatibilityChangeReporter: Compat change id reported: 184838306; UID 10145; state: DISABLED
04-27 14:22:01.149  1528  2453 D CompatibilityChangeReporter: Compat change id reported: 185004937; UID 10145; state: DISABLED
04-27 14:22:01.149  1528  2453 D CompatibilityChangeReporter: Compat change id reported: 181136395; UID 10145; state: DISABLED
04-27 14:22:01.149  1528  2453 D CompatibilityChangeReporter: Compat change id reported: 174042936; UID 10145; state: DISABLED
04-27 14:22:01.150  1528  2453 D CompatibilityChangeReporter: Compat change id reported: 197654537; UID 10145; state: DISABLED
04-27 14:22:01.151  1528  2453 V SplashScreenExceptionList: SplashScreen checking exception for package fr.free.nrw.commons (target sdk:25) -> false
04-27 14:22:01.151  1528  2453 D CompatibilityChangeReporter: Compat change id reported: 205907456; UID 10145; state: DISABLED
04-27 14:22:01.154  1528  2453 D CompatibilityChangeReporter: Compat change id reported: 194833441; UID 10145; state: DISABLED
04-27 14:22:01.158  1528  1728 D CompatibilityChangeReporter: Compat change id reported: 135634846; UID 10145; state: DISABLED
04-27 14:22:01.159  1528  1728 D CompatibilityChangeReporter: Compat change id reported: 177438394; UID 10145; state: DISABLED
04-27 14:22:01.159  1528  1728 D CompatibilityChangeReporter: Compat change id reported: 135772972; UID 10145; state: DISABLED
04-27 14:22:01.159  1528  1728 D CompatibilityChangeReporter: Compat change id reported: 135754954; UID 10145; state: DISABLED
04-27 14:22:01.159  1528  1744 D CompatibilityChangeReporter: Compat change id reported: 143937733; UID 10145; state: DISABLED
04-27 14:22:01.169  1528  2135 D CompatibilityChangeReporter: Compat change id reported: 168419799; UID 10145; state: DISABLED
04-27 14:22:01.174  1528  2382 D CoreBackPreview: Window{7971a6 u0 Splash Screen fr.free.nrw.commons}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@b158f94, mPriority=0}
04-27 14:22:01.174  1528  1744 I ActivityManager: Start proc 5833:fr.free.nrw.commons/u0a145 for next-top-activity {fr.free.nrw.commons/fr.free.nrw.commons.auth.LoginActivity}
04-27 14:22:01.202     0     0 I binder  : 5833:5833 ioctl 40046210 7fd579d820 returned -22
04-27 14:22:01.517  1528  2453 I ActivityTaskManager: START u0 {cmp=fr.free.nrw.commons/.WelcomeActivity} from uid 10145
04-27 14:22:01.531  1528  2135 I ActivityTaskManager: START u0 {flg=0x4000000 cmp=fr.free.nrw.commons/.contributions.ContributionsActivity} from uid 10145
04-27 14:22:01.541  1528  5536 W ActivityTaskManager: Tried to set launchTime (0) < mLastActivityLaunchTime (369064)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: FATAL EXCEPTION: main
04-27 14:22:01.559  5833  5833 E AndroidRuntime: Process: fr.free.nrw.commons, PID: 5833
04-27 14:22:01.559  5833  5833 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons/fr.free.nrw.commons.contributions.ContributionsActivity}: java.lang.NullPointerException: fr.free.nrw.commons.CommonsApplication.activityInjector() returned null
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3641)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3778)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:138)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2303)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7884)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: Caused by: java.lang.NullPointerException: fr.free.nrw.commons.CommonsApplication.activityInjector() returned null
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at dagger.internal.Preconditions.checkNotNull(Preconditions.java:79)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at dagger.android.AndroidInjection.inject(AndroidInjection.java:56)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at fr.free.nrw.commons.contributions.ContributionsActivity.onCreate(ContributionsActivity.java:142)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8341)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8320)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3622)
04-27 14:22:01.559  5833  5833 E AndroidRuntime: 	... 12 more
04-27 14:22:01.563  1528  5862 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
04-27 14:22:01.563  1528  5536 W ActivityTaskManager:   Force finishing activity fr.free.nrw.commons/.contributions.ContributionsActivity
04-27 14:22:01.676  1528  2453 I ActivityManager: Process fr.free.nrw.commons (pid 5833) has died: fg  TOP 
```

**Comment by VinzRoosen** [comment]
Hi, can i work on this?

**Comment by VinzRoosen** [comment]
Hi, can anyone tell me where I can find the version 2.6 used in F-droid? It seems that the oldest version available on github is release 2.7.x, yet F-droid's version is 2.6. If possible, can someone give me some more information?

**Comment by dboehmer** [comment]
I am no expert here but F-Droid links to a source tarball for all available versions on https://f-droid.org/packages/fr.free.nrw.commons/

While looking I realized that F-Droid provides 2.6.0 as well as 4.0.3. I don’t know why 2.6.0 is marked as *suggested* and installed by default. I have just manually selected 4.0.3 and logins works fine now. I can see my contributions and look forward to updating via the app soon.

I don’t mind if you close this issue because it seems to apply only to the old version 2.6.0 from 2017 (!).
